core-api/src/main/java/com/optimizely/ab/event/internal/EventFactory.java

```diff
@@ -168,6 +168,15 @@ public LogEvent createConversionEvent(@Nonnull ProjectConfig projectConfig,
         List<Attribute> attributesList = new ArrayList<Attribute>();
 
         for (Map.Entry<String, ?> entry : attributes.entrySet()) {
+            //Ignore null value attribute
+            if (entry.getValue() == null ||
+                    (!(entry.getValue() instanceof String) &&
+                    !(entry.getValue() instanceof Integer) &&
+                    !(entry.getValue() instanceof Double) &&
+                    !(entry.getValue() instanceof Boolean))) {
```

Comment:
Just a suggestion, Usually don't like too many Not conditions. so why we can't convert it into one not. 
!(e.getValue() instanceof String || e.getValue() instanceOf Int .... ) should be equal to the above condition.
@nchilada what do u suggest?

Reply:
Seems reasonable to me!